### PR TITLE
Move file backend to its own service

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Move file backend to its own service, the mailservice was a collateral
need.

### CODE REVIEW

File: services/mailservice/mail_test.go [hunk 1]
```diff
@@ -91,26 +93,26 @@ func TestMailConnectionAdvanced(t *testing.T) {
...
-	T = GetUserTranslations("en")
+	utils.T = utils.GetUserTranslations("en")
 
 	var emailTo = "test@example.com"
 	var emailSubject = "Testing this email"
 	var emailBody = "This is a test from autobot"
 
 	//Delete all the messages before check the sample email
-	DeleteMailBox(emailTo)
+	utils.DeleteMailBox(emailTo)
```
Comment (hmhealey) [inline services/mailservice/mail_test.go]: Can `utils/inbucket.go` be moved into this package as well and be marked as test code? I don't think it's used outside of testing this package

Comment (jespino) [reply services/mailservice/mail_test.go]: Good point, I going to take a look into it.

Comment (jespino) [reply services/mailservice/mail_test.go]: It is used in the tests of the `api4/team_test.go` and `api4/user_test.go`. We still can move it to a `mailservicetestutils` package or something like that inside the `mailservice` directory. and Import it from wherever is needed.

Comment (jespino) [reply services/mailservice/mail_test.go]: Done! :)

**Pull request merged**